src/joystick/hidapi/SDL_hidapi_switch.c

```diff
@@ -1107,9 +1179,51 @@ HIDAPI_DriverSwitch_SetJoystickLED(SDL_HIDAPI_Device *device, SDL_Joystick *joys
 }
 
 static int
-HIDAPI_DriverSwitch_SendJoystickEffect(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, const void *data, int size)
+HIDAPI_DriverSwitch_SendJoystickEffect(SDL_HIDAPI_Device *device, SDL_Joystick *joystick, const void *vibration, int size)
 {
-    return SDL_Unsupported();
+    SDL_DriverSwitch_Context *ctx = (SDL_DriverSwitch_Context *)device->context;
+
+    if (size != 8) {
+        SDL_SetError("Wrong size of vibration data");
+        return -1;
+    }
+    Uint16 vibration_data;
+    SDL_memcpy(&vibration_data, vibration, sizeof(vibration_data));
```

Comment:
You're only memcpying 2 bytes, not the 8 that are passed in, and then you're not using vibration_data. Am I missing something?

Reply:
Sorry. I though I have pushed my latest changes. Now it should be correct and functional